### ISSUE TITLE
How to Access LassoNet Model Wieghts

### ISSUE THREAD

**Issue by imessien** [lasso-net/lassonet#46]
I'm trying to pull the model weights selected in the final optimized regularization path for LassoNet. Is there anyway to show the weights for each of the features used in that path? I know you can print out the most complex path using path[-1] but how do I access the weights there? I've tried using path[-1].state_dict() but that didn't work. Is there a special way to call the weights in the path?  I know you can use model.feature_importance_ to show the feature importance but that doesn't serve as the weight in the model. Correct? The model.coef_ attribute doesn't work just to print out the weight. Could you please provide some guidance on how to obtain the weights or coefficients of the LassoNet model? Thanks for your help.

**Comment by louisabraham** [comment]
Can you provide the full code? The answer is not the same depending on the method you called. For a CV model, you do `model.path_[-1].state_dict`.

**Comment by imessien** [comment]
I was using the LassoNetRegressorCV. Thank you for the suggestion it worked that way. My goal is to create a scatter plot with all the individual features and show the final optimal path taken by connecting them. I want to annotate the top 10 features on this line. I want to shade the area underneath the line to show the selected features region. This area would represent the selection of features most likely used in the model but not the top 10. The points above the line weren't included in the final path taken by the model. One axis would be weights and I can't figure out what the other one would be. I think accuracy would be with respect to the output of the model. I want to give it weight  I'm confused about what my axis should be can it be accuracy and coefficients? When I use lambda for x and coefficients and y it would be a vertical line because the best features all have the same lambda value.  What should my axis be and is this possible? I will use coefficients = model.path_[-1].state_dict()['linear.weight'].numpy() to get the weights of each of the features. Is this possible?

**Comment by louisabraham** [comment]
You should be able to access it with just 

```
model.path_[-1].state_dict['linear.weight'].numpy()
```

**Comment by imessien** [comment]
That method gave me an error message. The correct statement is model.path_[-1].state_dict. 'linear.wieght' is not a key for the state_dict. The keys are ['layers.0.weight', 'layers.0.bias', 'layers.1.weight', 'layers.1.bias', 'skip.weight']. So I will use 'layers.0.weight' to get it. Thank you for your help.
